conformance/lang/expressions/new-expr/explicit-new-expr/stream_class_descriptor.balt

```diff
@@ -0,0 +1,303 @@
+    io:println(a); // @output 1
+    int|error? b = getRecordValue(numberStream.next());
+    io:println(b); // @output 2
+    error? closed = numberStream.close();
+    io:println(closed is ()); // @output true
+}
+
+Test-Case: panic
+Description: Test closed stream next.
+// Fail-Issue: ballerina-platform/ballerina-lang/issues/34934
+Labels: int, error, optional-type, new-expr, var, stream-type-descriptor
+
+function init() {
+    stream<int, error?> newStream = new stream<int, error?>();
+    error? closed = newStream.close();
+    var _ = newStream.next(); // @panic
+}
+
+Test-Case: panic
+Description: Test auto closing of stream after completion type.
+// Fail-Issue: ballerina-platform/ballerina-lang/issues/34934
+Labels: int, record-type, optional-type, if-else-stmt, is-expr, module-type-defn, module-class-defn, new-expr, stream-type-descriptor
+
+type IntRecord record {|
+    int value;
+|};
+
+function getRecordValue(IntRecord? returnedVal) returns int? {
+    if returnedVal is IntRecord {
+        return returnedVal.value;
+    } else {
+        return returnedVal;
+    }
+}
+
+class NumbersGenerator {
+    int i = 0;
+    public isolated function next() returns IntRecord? {
+        self.i += 1;
+        if (self.i > 1) {
+            self.i = 0;
+            return ();
+        }
+        return {value: self.i};
+    }
+}
+
+function init() {
+    NumbersGenerator numbersGenerator = new;
+    stream<int> newStream = new stream<int>(numbersGenerator);
+    var value = getRecordValue(newStream.next());
+    io:println(value); // @output 1
+    value = getRecordValue(newStream.next());
+    io:println(value is ()); // @output true
+    IntRecord? panicVar = newStream.next(); // @panic
+    io:println(panicVar is IntRecord); // @output false
+}
+
+Test-Case: panic
+Description: Test closed stream next when stream implementor does not have close method.
+// Fail-Issue: ballerina-platform/ballerina-lang/issues/34934
+Labels: int, record-type, error, optional-type, union-type, is-expr, module-type-defn, module-class-defn, explicit-new-expr, stream-type-descriptor
+
+type IntRecord record {|
+    int value;
+|};
+
+class NumbersGenerator {
+    int i = 0;
+    public isolated function next() returns IntRecord|error? {
+        self.i += 1;
+        return { value: self.i };
+    }
+}
+
+function init() {
+    NumbersGenerator numbersGenerator = new;
+    stream<int, error?> newStream = new stream<int, error?>(numbersGenerator);
+    error? closed = newStream.close();
+    io:println(closed is ()); // @output true
+    IntRecord|error? panicVar = newStream.next(); // @panic
+    io:println(panicVar is IntRecord); // @output false
+}
+
+Test-Case: output
+Description: Test closed stream next with stream implementor with close method.
+Labels: int, record-type, error, optional-type, union-type, module-type-defn, module-class-defn, if-else-stmt, is-expr, explicit-new-expr, stream-type-descriptor
+
```

Comment:
`return-stmt`, `nil-type`, `field-access-expr` labels are missing.

Reply:
Thanks. As of now `return-stmt`, `function-defn`, `function-call-expr` all are in the minimal subset and I am not testing them specifically. 
I added `nil-type`, `method-call-expr`, `member-access-expr`